providers/provider_data.go

```diff
@@ -17,6 +19,44 @@ type ProviderData struct {
 	ValidateURL       *url.URL
 	Scope             string
 	ApprovalPrompt    string
+	ExportingClaims   ExportingClaims
+}
+
+// AddressClaims represents the ODIC formatted address claim
+// https://openid.net/specs/openid-connect-core-1_0.html#AddressClaim
+type AddressClaims struct {
+	Country       string `json:"country"`
+	Formatted     string `json:"formatted"`
+	Locality      string `json:"locality"`
+	PostalCode    string `json:"postal_code"`
+	Region        string `json:"region"`
+	StreetAddress string `json:"street_address"`
+}
+
+// StandardClaims represents the OIDC core claims
+// https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims
+type StandardClaims struct {
```

Comment:
Is it possible there is an OIDC library we can import these from? We already import one right? (Same for AddressClaims)

Reply:
I couldn't find one, I will look into it again. go-oidc seems to be the library to use here, and they just grab a subset and leave the rest in a byte array. I tried using go-oidc's stuff a bit, but it got a bit messy trying to include it. I could see a PR where the providers implement go-oidc's interface instead of the custom one oauth2_proxy has.

Though I liked the well-defined struct behavior more, I ended up updating this to use map[string]interface{} claims unmarshalling because it allowed OIDC provider to unpack non-standard claims at will. I'll have an update for these changes in the next day or so.

Reply:
Feel free to push back on this if you've already tried my suggestion and you feel like this is a better approach!

OSS is a democracy 😛

Reply:
Thank you for saying that. I don't know yet. I'm trying a few variations to see what works or doesn't.